### DISCUSSION TITLE
codeLens still creates stylesheet when disabled

### DISCUSSION THREAD

**Discussion by KamranAsif** [microsoft/monaco-editor#3746] [Q&A] [answered]
**monaco-editor version:** 0.25.1
**Browser:** Chrome
**OS:** Mac
**Playground code that reproduces the issue:**

```js
monaco.editor.create(document.getElementById("container"), {
    codeLens: false,
    value: "function hello() {\n\talert('Hello world!');\n}",
    language: "javascript"
});

const hasCodelensStyles =
    Array.from(document.querySelectorAll('style[media="screen"]'))
        .some(style => style.textContent.includes('codelens'));
alert(hasCodelensStyles)
```

If codeLens is disabled, monaco editor should not initialize the stylesheet. This can be traced back to this line here: https://github.com/microsoft/vscode/blob/2e165500a6d5449fdd076f0ddbaa588f1e5743c7/src/vs/editor/contrib/codelens/codelensController.ts#L67

**Comment by KamranAsif** [comment]
Would even accept a workaround where it would disable adding just the style sheets

**Comment by alexdima** [comment, marked as answer]
With the [Monaco Editor Webpack Plugin](https://github.com/microsoft/monaco-editor-webpack-plugin) it is possible to exclude the codelens.ts file completely if you don't want that feature.
